### PR TITLE
Get a (somewhat) working cluster

### PR DESCRIPTION
**Pull request by Timdnb**

### CODE REVIEW

**Comment by Timdnb** [comment]
When you want to use kubectl from your own host, do the following

go to the k3s.yaml file and change "server: https://127.0.0.1:6443" -> "server: https://192.168.56.10:6443" and then run
export KUBECONFIG=./playbooks/k3s.yaml

When you want to use kubectl within the nodes, do the following

go to the k3s.yaml file and change "server: https://127.0.0.1:6443" -> "server: https://192.168.56.10:6443" and then run
export KUBECONFIG=/vagrant/playbooks/k3s.yaml

File: kubernetes/ingress.yml [hunk 1]
```diff
@@ -1,10 +1,18 @@
 apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
-  name: ingress-app
+  name: app-ingress
+  annotations:
+    nginx.ingress.kubernetes.io/rewrite-target: /
```
Comment (SagaRut) [inline kubernetes/ingress.yml:6]: I thought we bailed on nginx

File: kubernetes/ingress.yml [hunk 2]
```diff
@@ -1,10 +1,18 @@
...
+  - host: app.local
+    http:
+      paths:
+      - path: /
+        pathType: Prefix
+        backend:
+          service:
+            name: service-app
+            port:
+              number: 5000
```
Comment (SagaRut) [inline kubernetes/ingress.yml:18]: Do we really need these changes

File: playbooks/setup_k8s_cluster.yml [hunk 3]
```diff
@@ -59,47 +41,6 @@
...
-    - name: Export config
-      shell: |
-        export KUBECONFIG=~/.kube/config
-
-    - name: Apply Kubernetes yml files
-      shell: |
-        kubectl apply -f ./../vagrant/kubernetes/environment.yml
-        kubectl apply -f ./../vagrant/kubernetes/app.yml
-        kubectl apply -f ./../vagrant/kubernetes/model-service.yml
-        kubectl apply -f ../../vagrant/kubernetes/ingress.yml
```
Comment (SagaRut) [inline playbooks/setup_k8s_cluster.yml:105]: Do we really want to exclude this from the playbooks?

File: kubernetes/ingress.yml (hunk 1 above)
Comment (Timdnb) [reply kubernetes/ingress.yml:6]: well yes, never actually noticed it says nginx haha, but this is also just the default ingress template from kubernetes (see [this](https://kubernetes.io/docs/concepts/services-networking/ingress/)). and it works in this configuration

File: kubernetes/ingress.yml (hunk 2 above)
Comment (Timdnb) [reply kubernetes/ingress.yml:18]: see comment above, it's just the default template for an ingress

**Comment by Timdnb** [comment]
> > When you want to use kubectl from your own host, do the following
> > go to the k3s.yaml file and change "server: https://127.0.0.1:6443" -> "server: https://192.168.56.10:6443" and then run export KUBECONFIG=./playbooks/k3s.yaml
> > When you want to use kubectl within the nodes, do the following
> > go to the k3s.yaml file and change "server: https://127.0.0.1:6443" -> "server: https://192.168.56.10:6443" and then run export KUBECONFIG=/vagrant/playbooks/k3s.yaml
> 
> Wait since when do we need to do this...?

Since I took out all the somewhat unnecessary steps that were in the playbooks, as in a real case you would control the cluster from outside of it, and before it was setup that you can access it everywhere. But it's the same as before, the k3s.yaml just needs the correct IP from the api server and the KUBECONFIG is the same, just different file location

**Comment by Timdnb** [comment]
I think I can automate the manual change you now need to do in the k3s.yaml, ill have a look

File: playbooks/setup_k8s_cluster.yml (hunk 3 above)
Comment (SagaRut) [reply playbooks/setup_k8s_cluster.yml:105]: Yeah I guess, but then we need to update the readme what exactly needs to be applied and installed

Comment (SagaRut) [reply playbooks/setup_k8s_cluster.yml:105]: yeah but also just for us

Comment (Timdnb) [reply playbooks/setup_k8s_cluster.yml:105]: ill put it back then

Comment (Timdnb) [reply playbooks/setup_k8s_cluster.yml]: too late ;p, ill still add the above to the README for now, but in the future a user should not have to do that

**Review by SagaRut — APPROVED** [review]

**Pull request merged**